idunn/blocks/wikipedia.py

```diff
@@ -100,6 +101,50 @@ def get_title_in_language(self, title, source_lang, dest_lang):
 
         return None
 
+class Wikidata:
+    _wiki_es = None
+    _es_lang = None
+
+    @classmethod
+    def init_lang(cls):
+        from app import settings
+        cls._es_lang = settings['ES_WIKI_LANG'].split(',')
+
+    @classmethod
+    def get_wiki_index(cls, lang):
+        if cls._es_lang is None:
+            cls.init_lang()
+        if lang in cls._es_lang:
+            return "wikidata_{}".format(lang)
+        return None
+
+    @classmethod
+    def get_wiki_es(cls):
+        if cls._wiki_es is None:
+            from app import settings
+            cls._wiki_es = Elasticsearch(settings['WIKI_ES'])
+        return cls._wiki_es
+
+    @classmethod
+    def get_wiki_info(cls, wikidata_id, lang, wiki_index):
+
+        resp = cls.get_wiki_es().search(
+            index=wiki_index,
+            body={
+                "filter": {
+                    "term": {
+                        "wikibase_item": wikidata_id
+                    }
+                }
+            }
+        ).get('hits', {}).get('hits', [])
+
+        if len(resp) == 0:
+            return None
+
+        wiki = resp[0]['_source']
+        return (wiki.get("url"), wiki.get("title"), wiki.get("content"))
```

Comment:
I'm not a fan of returning complex tuples, I'd prefer an object

Reply:
done